activerecord/lib/active_record/reflection.rb

```diff
@@ -623,7 +623,11 @@ def calculate_constructable(macro, options)
         # +nil+.
         def inverse_name
           options.fetch(:inverse_of) do
-            @automatic_inverse_of ||= automatic_inverse_of
+            if @automatic_inverse_of == false
+              nil
+            else
+              @automatic_inverse_of ||= automatic_inverse_of
```

Comment:
If `automatic_inverse_of` returns `false`, the else path will return `false` at least once.
So how about here?

```diff
--- a/activerecord/lib/active_record/reflection.rb
+++ b/activerecord/lib/active_record/reflection.rb
@@ -425,7 +425,6 @@ def compute_class(name)
 
       def initialize(name, scope, options, active_record)
         super
-        @automatic_inverse_of = nil
         @type         = options[:as] && (options[:foreign_type] || "#{options[:as]}_type")
         @foreign_type = options[:foreign_type] || "#{name}_type"
         @constructable = calculate_constructable(macro, options)
@@ -610,11 +609,12 @@ def calculate_constructable(macro, options)
         # +nil+.
         def inverse_name
           options.fetch(:inverse_of) do
-            @automatic_inverse_of ||= automatic_inverse_of
+            @inverse_name = automatic_inverse_of unless defined?(@inverse_name)
+            @inverse_name
           end
         end
 
-        # returns either false or the inverse association name that it finds.
+        # returns either +nil+ or the inverse association name that it finds.
         def automatic_inverse_of
           if can_find_inverse_of_automatically?(self)
             inverse_name = ActiveSupport::Inflector.underscore(options[:as] || active_record.name.demodulize).to_sym
@@ -631,8 +631,6 @@ def automatic_inverse_of
               return inverse_name
             end
           end
-
-          false
         end
```

Reply:
Thanks for the suggestion :+1:  I've updated the PR.